homeassistant/components/upcloud/binary_sensor.py

```diff
@@ -16,14 +18,19 @@
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
Home Assistant has a built-in warning for when this method does not exist. So you can just drop this method.

Reply:
Ok, but if you don't feel strongly about this, I'd rather keep this around as it provides a more specific/accurate info and somewhat more "polite" tone:
> WARNING: Loading upcloud binary sensors via platform config is deprecated and no longer necessary as of 0.117. Please remove it from binary_sensor YAML configuration.

vs

> ERROR: The upcloud platform for the binary_sensor integration does not support platform setup. Please remove it from your config.

(Aside, the generic message might want to explicitly mention _YAML_ config.)

Reply:
We should update our built-in warning instead of adding a new one because it's more polite or mentions YAML.

Reply:
Politeness and YAML aside, ability to show the versioning info would likely result in about as much code needed in an integration as just implementing the function like was done here. Not sure what could be done about the distinction between error/warning in a generic way. But I'll just drop the function here so we can move on.

Reply:
Dropped from here and from switch.